src/http/client.cc

```diff
@@ -70,17 +70,17 @@ future<> connection::write_body(request& req) {
     }
 }
 
-future<std::optional<reply>> connection::maybe_wait_for_continue(request& req) {
+future<connection::reply_ptr> connection::maybe_wait_for_continue(request& req) {
     if (req.get_header("Expect") == "") {
-        return make_ready_future<std::optional<reply>>(std::nullopt);
+        return make_ready_future<reply_ptr>(nullptr);
     }
 
     return _write_buf.flush().then([this] {
-        return recv_reply().then([] (reply rep) {
-            if (rep._status == reply::status_type::continue_) {
-                return make_ready_future<std::optional<reply>>(std::nullopt);
+        return recv_reply().then([] (reply_ptr rep) {
+            if (rep->_status == reply::status_type::continue_) {
```

Comment:
Note: Here you can see the downside of always using the unique_ptr: Here you **know** that rep is non-null (because recv_reply() never returns null), so you can use `rep->...` without checking, but the compiler can't help you verify you don't make mistake. It's almost like using std::optional<> everywhere even on something you know isn't optional.